keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -867,28 +873,13 @@ class SqlArtifactRepository(
   override fun getArtifactSummaryInEnvironment(
     deliveryConfig: DeliveryConfig,
     environmentName: String,
-    artifactName: String,
-    artifactType: ArtifactType,
+    artifactReference: String,
     version: String
   ): ArtifactSummaryInEnvironment? {
     return sqlRetry.withRetry(READ) {
 
```

Comment:
There were 3 sql queries in this function. I removed these two, and added that sql into the third query. Then I added a nested query to get the pinned information. I'd love ideas on how I could remove the nested query.

Basically, I want to have a `true` or `false` if there's a row that matches. This will be used to populate the `pinned` field.

Reply:
> There were 3 sql queries in this function. I removed these two, and added that sql into the third query. Then I added a nested query to get the pinned information. I'd love ideas on how I could remove the nested query.

Yeah, I know they weren't the most efficient approach, but I struggle reading nested or really large queries, and nesting also makes it impossible to throw specific errors like I did here, so that was on purpose. 😄

Also, the things you replaced them with both cause queries to be executed, so in terms of performance it's the same. Having said that, it does make things shorter, which might be easier to read, so I don't have any major objections, just wanted to explain what my rationale was.

Reply:
Oooh it still executes them like that? Darn!! Ok I'll have to rethink this. Thanks.

I just want to speed the generation of this view up, because it's definitely getting slow. I struggle to read the big queries too, but I think it's worth it because making multiple sql connections is slower.

Reply:
Actually, the functions I used just generate other sql and embed it in the query. So I think it is reducing the number of connections to sql.

Reply:
FWIW, I don't think connecting to the database is the problem, but rather executing queries. Jooq uses a pool of reusable connections.